drivers/gles2/rasterizer_canvas_base_gles2.cpp

```diff
@@ -250,7 +250,8 @@ void RasterizerCanvasBaseGLES2::draw_window_margins(int *black_margin, RID *blac
 	if (black_image[MARGIN_LEFT].is_valid()) {
 		_bind_canvas_texture(black_image[MARGIN_LEFT], RID());
 		Size2 sz(storage->texture_get_width(black_image[MARGIN_LEFT]), storage->texture_get_height(black_image[MARGIN_LEFT]));
-		draw_generic_textured_rect(Rect2(0, 0, black_margin[MARGIN_LEFT], window_h), Rect2(0, 0, sz.x, sz.y));
+		draw_generic_textured_rect(Rect2(0, 0, black_margin[MARGIN_LEFT], window_h),
+				Rect2(0, 0, (float)black_margin[MARGIN_LEFT] / sz.x, (float)(window_h) / sz.y));
```

Comment:
Are we confident that `sz.x` and `sz.y` can't ever be 0?

Reply:
I tested with this and it did not crash nor return any error, so I guess there's some validation that happens further up the stack:
```gdscript
	var rid = VisualServer.texture_create()
	print(rid.get_id())  # 147
	print(VisualServer.texture_get_height(rid))  # 0
	print(VisualServer.texture_get_width(rid))  # 0
	
	VisualServer.black_bars_set_images(rid, rid, rid, rid)
	VisualServer.black_bars_set_margins(50,50,50,50)
```